controllers/register_test.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/gutakk/go-google-scraper/db"
+	"github.com/gutakk/go-google-scraper/models"
+	"github.com/gutakk/go-google-scraper/tests"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/go-playground/assert.v1"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+)
+
+func TestDisplayRegister(t *testing.T) {
+	engine := tests.GetRouter(true)
+	new(RegisterController).applyRoutes(engine)
+
+	response := tests.PerformRequest(engine, "GET", "/register", nil, nil)
+	p, err := ioutil.ReadAll(response.Body)
+	pageOK := err == nil && strings.Index(string(p), "<title>Register</title>") > 0
+
+	assert.Equal(t, http.StatusOK, response.Code)
+	assert.Equal(t, true, pageOK)
+}
+
+type DBTestSuite struct {
+	suite.Suite
+	engine   *gin.Engine
+	formData url.Values
+	headers  http.Header
+	email    string
+	password string
+}
+
+func (s *DBTestSuite) SetupTest() {
+	testDB, _ := gorm.Open(postgres.Open(tests.ConstructTestDsn()), &gorm.Config{})
+	db.GetDB = func() *gorm.DB {
+		return testDB
+	}
+
+	_ = db.GetDB().AutoMigrate(&models.User{})
+
+	s.engine = tests.GetRouter(true)
+	registerController := &RegisterController{}
+	registerController.applyRoutes(s.engine)
+
+	s.headers = http.Header{}
+	s.headers.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	s.email = faker.Email()
+	s.password = faker.Password()
+
+	s.formData = url.Values{}
+	s.formData.Set("email", s.email)
+	s.formData.Set("password", s.password)
+	s.formData.Set("confirm-password", s.password)
+}
+
+func (s *DBTestSuite) TearDownTest() {
+	db.GetDB().Exec("DELETE FROM users")
+}
+
+func TestDBTestSuite(t *testing.T) {
+	suite.Run(t, new(DBTestSuite))
+}
+
+func (s *DBTestSuite) TestRegisterWithValidParameters() {
+	response := tests.PerformRequest(s.engine, "POST", "/register", s.headers, s.formData)
+
+	assert.Equal(s.T(), http.StatusFound, response.Code)
+	assert.Equal(s.T(), "/", response.Header().Get("Location"))
+}
+
+func (s *DBTestSuite) TestRegisterWithBlankEmailValidation() {
+	s.formData.Del("email")
+
+	response := tests.PerformRequest(s.engine, "POST", "/register", s.headers, s.formData)
+	p, err := ioutil.ReadAll(response.Body)
+	pageError := err == nil && strings.Index(string(p), "Invalid email format") > 0
+
+	assert.Equal(s.T(), http.StatusBadRequest, response.Code)
+	assert.Equal(s.T(), true, pageError)
+}
+
+func (s *DBTestSuite) TestRegisterWithBlankPasswordValidation() {
+	s.formData.Del("password")
+
+	response := tests.PerformRequest(s.engine, "POST", "/register", s.headers, s.formData)
+	p, err := ioutil.ReadAll(response.Body)
+	pageError := err == nil && strings.Index(string(p), "Password is required") > 0
+	isEmailFieldValueExist := err == nil && strings.Index(string(p), s.email) > 0
+
+	assert.Equal(s.T(), http.StatusBadRequest, response.Code)
+	assert.Equal(s.T(), true, pageError)
+	assert.Equal(s.T(), true, isEmailFieldValueExist)
+}
+
+func (s *DBTestSuite) TestRegisterWithPasswordNotMatchValidation() {
+	s.formData.Set("confirm-password", "1234567")
```

Comment:
Minor thing but for invalid data, make it easy to reach that it's invalid without having to check the sample data created above:

```suggestion
	s.formData.Set("confirm-password", "invalid")
```

Reply:
Fixed in 99c62793ff03294f062aee9677b11fd77b98247d